XIPs/xip-10-brandinfo-schema.md

```diff
@@ -0,0 +1,90 @@
+---
+xip: 14
+title: Conversation context metadta schema
+description: Schema to enable standardized sharing of conversation context
+author: Yash Lunagaria (@yash-luna)
+status: Review
+type: Standards Track
+category: XRC
+created: 2023-01-31
+---
+
+## Abstract
+
+This XIP proposes a schema for sharing additional context about a conversation beyond a `conversationId`. Having a standard way to provide not only a conversationId, but also conversation metadata will enable different frontends to display useful contextual information in a consistent, user-friendly manner, while still affording apps design and UI flexibility.
+
+## Motivation
+
+The SDK API currently accepts a unique `conversationId` per address pair and allows `metadata` to be set as well. Today, different apps use these parameters in different ways. Therefore an app may not know how to correctly render in its UI `conversationId` and `metadata` for conversations originating from other apps. We propose a schema to standardize the use of conversation `metadata` for the purpose of app brand expression stemming from the `conversationId`. The `metadata` field may include other non-standard (or future standard) properties.
+
+Note that the schema is only applicable when setting a non-null `conversationId`. If `conversationId` is null, the SDK does not allow setting `metadata`.
+
+## Specification
+
+Proposed `displayInfo` schema in conversation `metadata`
+
+```json
+{
+    conversationId: "mydomain.xyz/abc/qrs",
+    metadata: { 
+        "displayInfo.prettyName": "My company", 
+        "displayInfo.profileImage": "mydomain.xyz/assets/myimage.png", 
+        "displayInfo.primaryColor": "#ffffff" }
+}
+```
+
+Example `displayInfo` implementation for a chat app named Galaxy
+
+```json
+{
+    conversationId: "galaxy.chat/dm/uniqueIdentifier",
+    metadata: { 
+        "displayInfo.prettyName": "Galaxy", 
+        "displayInfo.profileImage": "galaxychat.xyz/brandassets/logo.png", 
```

Comment:
What is the rationale behind embedding this data into the Conversation Invites? 

Why does the invite not contain a pointer to a central registry, which would be easier for updates?
Why is this information not contained in a metamessage within the conversation?

Reply:
I think these are valid reasons to embedded into the conversation metainfo.  I can get behind this. 

In my mind a central registry could be ENS with values stored as textfields. It would address Longterm availability (at least in the context of apps), and Ease of use (Passing update costs to the brands/applications). I think consistency will be an issue regardless and a well defined spec is needed.

Reply:
@richardhuaaa agree on the registry component for sure.

I do wonder if somehow the metadata itself can be hashed, and the hash compared against the registry, to know what's authentic. And perhaps the conversationId could be used as a checksum against the metadata, like how you might with a barcode.